From: Holger Assmann <h.assmann@pengutronix.de>
Date: Mon, 12 Jul 2021 11:21:33 +0200
Subject: [PATCH] lxatac: introduce IOBUS_FLT_FB

Overwrite original (but not needed) ethernet functionality of the pad in
order to use it as an GPIO.

Signed-off-by: Holger Assmann <h.assmann@pengutronix.de>
---
 arch/arm/boot/dts/stm32mp157c-lxa-tac.dts | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/stm32mp157c-lxa-tac.dts b/arch/arm/boot/dts/stm32mp157c-lxa-tac.dts
index fbbcb95f4903..5d93b520fb85 100644
--- a/arch/arm/boot/dts/stm32mp157c-lxa-tac.dts
+++ b/arch/arm/boot/dts/stm32mp157c-lxa-tac.dts
@@ -68,6 +68,7 @@ reg_1v2: regulator_1v2 {
 		regulator-max-microvolt = <1200000>;
 		regulator-always-on;
 	};
+
 };
 
 /* VREFBUF: What about the external reference voltage? */
@@ -102,6 +103,12 @@ &gpioe {
 	"", "", "", "", "UART_TX_EN", "UART_RX_EN", "", "", "", "", "";
 };
 
+&gpiog {
+	gpio-line-names = "", "", "", "", "", \
+	"IOBUS_FLT_FB", "", "", "", "", "", "", "", "", "", "";
+};
+
+
 &gpioh {
 	gpio-line-names = "", "", "OUT_1", "OUT_0", "", \
 	"", "", "", "", "", "", "", "", "IOBUS_PWR_EN", "", "";
@@ -324,8 +331,7 @@ pins {
 	};
 	ethernet0_rgmii_pins_d: rgmii-1 {
 		pins1 {
-			pinmux = <STM32_PINMUX('G', 5, AF11)>, /* ETH_RGMII_CLK125 */
-				 <STM32_PINMUX('G', 4, AF11)>, /* ETH_RGMII_GTX_CLK */
+			pinmux = <STM32_PINMUX('G', 4, AF11)>, /* ETH_RGMII_GTX_CLK */
 				 <STM32_PINMUX('G', 13, AF11)>, /* ETH_RGMII_TXD0 */
 				 <STM32_PINMUX('G', 14, AF11)>, /* ETH_RGMII_TXD1 */
 				 <STM32_PINMUX('C', 2, AF11)>, /* ETH_RGMII_TXD2 */
@@ -347,8 +353,7 @@ pins2 {
 	};
 	ethernet0_rgmii_sleep_pins_d: rgmii-sleep-1 {
 		pins1 {
-			pinmux = <STM32_PINMUX('G', 5, ANALOG)>, /* ETH_RGMII_CLK125 */
-				 <STM32_PINMUX('G', 4, ANALOG)>, /* ETH_RGMII_GTX_CLK */
+			pinmux = <STM32_PINMUX('G', 4, ANALOG)>, /* ETH_RGMII_GTX_CLK */
 				 <STM32_PINMUX('G', 13, ANALOG)>, /* ETH_RGMII_TXD0 */
 				 <STM32_PINMUX('G', 14, ANALOG)>, /* ETH_RGMII_TXD1 */
 				 <STM32_PINMUX('C', 2, ANALOG)>, /* ETH_RGMII_TXD2 */
